src/go/wsl-helper/cmd/kubeconfig.go

```diff
@@ -47,70 +45,82 @@ var kubeconfigCmd = &cobra.Command{
 	Long:  `This command configures the Kubernetes configuration inside a WSL2 distribution.`,
 	Args:  cobra.ExactArgs(0),
 	RunE: func(cmd *cobra.Command, args []string) error {
-		winConfigPath := kubeconfigViper.GetString("kubeconfig")
-		linuxConfigDir := filepath.Join(homedir.HomeDir(), ".kube")
-		linuxConfigPath := filepath.Join(linuxConfigDir, "config")
+		configPath := kubeconfigViper.GetString("kubeconfig")
 		enable := kubeconfigViper.GetBool("enable")
 
-		if winConfigPath == "" {
-			//lint:ignore ST1005 The capitalization is for a proper noun.
+		if configPath == "" {
 			return errors.New("Windows kubeconfig not supplied")
 		}
 
-		// Backwards compatibility: if the Linux config is a symlink to the Windows
-		// kubeconfig, unlink it.  This avoids issues where we clobber the Windows
-		// kubeconfig by accident.
-		if _, err := os.Readlink(linuxConfigPath); err == nil {
-			linuxInfo, err1 := os.Stat(linuxConfigPath)
-			windowsInfo, err2 := os.Stat(winConfigPath)
-			if err1 == nil && err2 == nil && os.SameFile(linuxInfo, windowsInfo) {
-				if err = os.Remove(linuxConfigPath); err != nil {
-					return fmt.Errorf("failed to remove kubeconfig symlink: %w", err)
-				}
-			}
-		}
-
-		if !enable {
-			return nil
-		}
-
-		cmd.SilenceUsage = true
-
-		winConfig, err := readKubeConfig(winConfigPath)
-		if err != nil {
-			return err
-		}
-
-		linuxConfig, err := readKubeConfig(linuxConfigPath)
-		if err != nil && !errors.Is(err, os.ErrNotExist) {
-			return err
-		}
-
-		cleanConfig := removeExistingRDConfig(rdCluster, &linuxConfig)
-
-		kubeConfig, err := updateKubeConfig(winConfig, *cleanConfig, rdNetworking)
+		_, err := os.Stat(configPath)
 		if err != nil {
-			return fmt.Errorf("failed to construct kubeconfig: %w", err)
+			return fmt.Errorf("could not open Windows kubeconfig: %w", err)
 		}
+		cmd.SilenceUsage = true
 
-		var finalKubeConfigFile *os.File
-		if err := os.MkdirAll(linuxConfigDir, 0o750); err != nil {
-			return err
-		}
-		finalKubeConfigFile, err = os.Create(linuxConfigPath)
-		if err != nil {
-			return err
+		configDir := path.Join(homedir.HomeDir(), ".kube")
+		linkPath := path.Join(configDir, "config")
+		// Check to see if config is rancher desktop only
+		var rdIndex = 0
+		if exitingCfg, err := readKubeConfig(linkPath); err == nil {
+			// If the config contains a single entry for Contexts, Clusters and Users and if the
+			// name is equal to rancher desktop, we assume that this kubeConfig only contains RD config
+			if len(exitingCfg.Contexts) == 1 && exitingCfg.Contexts[rdIndex].Name == rdCluster &&
+				len(exitingCfg.Clusters) == 1 && exitingCfg.Clusters[rdIndex].Name == rdCluster &&
+				len(exitingCfg.Users) == 1 && exitingCfg.Users[rdIndex].Name == rdCluster {
+				if err := removeConfig(linkPath); err != nil {
```

Comment:
No need to declare a new `err`, we already have an `err` in scope.
```suggestion
				if err = removeConfig(linkPath); err != nil {
```